open-craft/edx-platform
# Enable AWS S3 encryption

Also updates django-storages to 1.5.2, replace its S3BotoStorage class with S3Boto3Storage, which uses the newer boto3 library instead of the older boto one. For consistency, also replace boto with boto3 in entire code, except in 2 Python modules, for the reasons explained in the corresponding commit.

Test instructions
- Create a OCIM VM with this branch and log in via SSH
- Run "sudo su"
- Update env JSON files
  - In /edx/app/edxapp/lms.env.json, update "class" attribute in PROFILE_IMAGE_BACKEND dict from "storages.backends.s3boto.S3BotoStorage" to "storages.backends.s3boto3.S3Boto3Storage" 
  - In /edx/app/edxapp/lms.env.json, update "STORAGE_CLASS" attribute in GRADES_DOWNLOAD dict from "storages.backends.s3boto.S3BotoStorage" to "storages.backends.s3boto3.S3Boto3Storage"
  - Update "DEFAULT_FILE_STORAGE" from "storages.backends.s3boto.S3BotoStorage" to "storages.backends.s3boto.S3Boto3Storage" in all env JSONs with the following command:
 cd /edx/app/edxapp; find -maxdepth 1 -name '*.json' | xargs sed -i 's/"DEFAULT_FILE_STORAGE": "storages.backends.s3boto.S3BotoStorage"/"DEFAULT_FILE_STORAGE": "storages.backends.s3boto3.S3Boto3Storage"/'
  - Include "STORAGE_CLASS": "django.core.files.storage.FileSystemStorage" in VIDEO_IMAGE_SETTINGS and VIDEO_TRANSCRIPTS_SETTINGS in lms.env.json and cms.env.json

- Restart LMS: /edx/bin/supervisorctl restart lms
- Restart CMS:  /edx/bin/supervisorctl restart cms
- In LMS, login, go to the user profile and try to update the user profile picture. 
- In LMS django admin web UI, make your user a staff member/admin; in LMS, open the demo course, go to the instructor's dashboard and try to generate all reports
- In CMS, open the demo course, select the "Tools" > "Export" option in the top menu and try to export the demo course.
- In CMS, create a discussion block in any part of the course. In LMS, access it and make sure you can create posts with image attachments.

LMS URL: https://<instance_name>.opencraft.hosting/
LMS django admin web UI URL: https://<instance_name>.opencraft.hosting/admin
CMS URL: https://studio.<instance_name>.opencraft.hosting/

TODOs:
- Update PROFILE_IMAGE_BACKEND, GRADES_DOWNLOAD, DEFAULT_FILE_STORAGE, VIDEO_IMAGE_SETTINGS and VIDEO_TRANSCRIPTS_SETTINGS  values in env JSONs creation
